src/main/cpp/src/SparkResourceAdaptorJni.cpp

```diff
@@ -838,7 +849,7 @@ class spark_resource_adaptor final : public rmm::mr::device_memory_resource {
     // addr is not used yet, but is here in case we want it in the future.
     // amount is not used yet, but is here in case we want it for debugging/metrics.
```

Comment:
nit: please update the comment here about amount, so it is now correct, or just delete the line.